kubeutils/kubeutils.go

```diff
@@ -0,0 +1,51 @@
+package kubeutils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/Azure/adx-automation-agent/common"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// IsInCluster returns True if the current environment is in a Kubernetes cluster
+func IsInCluster() bool {
+	_, exists := os.LookupEnv(common.EnvKeyStoreName)
+	return exists
+}
+
+// CreateKubeClientset creates a new kubernetes clientset
+func CreateKubeClientset() (clientset *kubernetes.Clientset, err error) {
+	var config *rest.Config
+
+	if IsInCluster() {
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, fmt.Errorf("Fail to create in cluster kubernetes config | Error %s", err)
+		}
+	} else {
+		kubeconfigPath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
```

Comment:
>os.Getenv("HOME") [](start = 34, length = 17)

This could be more idiomatic
[https://golang.org/pkg/os/user/#User](https://golang.org/pkg/os/user/#User)

Reply:
That's ... three more lines of `if err != nil`

Reply:
I see. If this is not expected to run on windows, no probs :)